plugins/geofence/geofence_impl.cpp

```diff
@@ -0,0 +1,219 @@
+                                   _mavlink_geofence_item_messages.size(),
+                                   MAV_MISSION_TYPE_FENCE);
+
+    LogDebug() << "About to send " << _mavlink_geofence_item_messages.size() << " geofence items";
+
+    if (!_parent->send_message(message)) {
+        report_geofence_result(callback, Geofence::Result::ERROR);
+    }
+
+    _result_callback = callback;
+
+    _parent->register_timeout_handler(
+        std::bind(&GeofenceImpl::timeout_happened, this), 2.0, &_timeout_cookie);
+}
+
+void GeofenceImpl::process_mission_request_int(const mavlink_message_t &message)
+{
+    if (!_active) {
+        LogDebug() << "Ignore geofence request, currently inactive";
+        return;
+    }
+
+    mavlink_mission_request_int_t mission_request_int;
+    mavlink_msg_mission_request_int_decode(&message, &mission_request_int);
+
+    if (mission_request_int.target_system != _parent->get_own_system_id() &&
+        mission_request_int.target_component != _parent->get_own_component_id()) {
+        LogDebug() << "Ignore geofence request int that is not for us";
+        return;
+    }
+
+    if (mission_request_int.mission_type != MAV_MISSION_TYPE_FENCE) {
+        LogDebug() << "Ignore mission request that is not a geofence";
+        return;
+    }
+
+    send_geofence_item(mission_request_int.seq);
+
+    // Reset the timeout because we're still communicating.
+    _parent->refresh_timeout_handler(this);
+}
+
+void GeofenceImpl::process_mission_ack(const mavlink_message_t &message)
+{
+    if (!_active) {
+        LogDebug() << "Ignore mission ack because not active";
+        return;
+    }
+
+    mavlink_mission_ack_t mission_ack;
+    mavlink_msg_mission_ack_decode(&message, &mission_ack);
+
+    if (mission_ack.target_system != _parent->get_own_system_id() &&
+        mission_ack.target_component != _parent->get_own_component_id()) {
+        LogDebug() << "Ignore geofence ack that is not for us";
+        return;
+    }
+
+    // We got some response, so it wasn't a timeout and we can remove it.
+    _parent->unregister_timeout_handler(this);
+
+    if (mission_ack.type == MAV_MISSION_ACCEPTED) {
+        report_geofence_result(_result_callback, Geofence::Result::SUCCESS);
+        LogDebug() << "Sucess, done";
+
+    } else if (mission_ack.type == MAV_MISSION_NO_SPACE) {
+        LogDebug() << "Error: too many geofence items: " << int(mission_ack.type);
+        report_geofence_result(_result_callback, Geofence::Result::TOO_MANY_GEOFENCE_ITEMS);
+    } else {
+        LogDebug() << "Error: unknown geofence ack: " << int(mission_ack.type);
+        report_geofence_result(_result_callback, Geofence::Result::ERROR);
+    }
+}
+
+void GeofenceImpl::assemble_mavlink_messages(
+    const std::vector<std::shared_ptr<Geofence::Polygon>> &polygons)
+{
+    // TODO: delete all entries first
+    _mavlink_geofence_item_messages.clear();
+
+    for (auto &polygon : polygons) {
+        u_int16_t command;
```

Comment:
What is `u_int16_t` supposed to be? It is apparently not defined on Windows, and from what I understand it is not standard. Why not `uint16_t` (if that solves the CI error)?

Reply:
I think it is a mistake - hopefully the newest commit fixes it